service/controller/resource/monitoring/remotewriteagentconfigsecret/resource.go

```diff
@@ -0,0 +1,117 @@
+package remotewriteagentconfigsecret
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/giantswarm/k8sclient/v7/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	pov1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/prometheus-meta-operator/v2/service/controller/resource/generic"
+	"github.com/giantswarm/prometheus-meta-operator/v2/service/key"
+)
+
+const (
+	Name = "remotewriteagentconfigsecret"
+)
+
+type Config struct {
+	K8sClient  k8sclient.Interface
+	Logger     micrologger.Logger
+	BaseDomain string
+}
+
+func New(config Config) (*generic.Resource, error) {
+	clientFunc := func(namespace string) generic.Interface {
+		c := config.K8sClient.K8sClient().CoreV1().Secrets(namespace)
+		return wrappedClient{client: c}
+	}
+
+	c := generic.Config{
+		ClientFunc:    clientFunc,
+		Logger:        config.Logger,
+		Name:          Name,
+		GetObjectMeta: getObjectMeta,
+		GetDesiredObject: func(ctx context.Context, v interface{}) (metav1.Object, error) {
+			return toSecret(ctx, v, config)
+		},
+		HasChangedFunc: hasChanged,
+	}
+	r, err := generic.New(c)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return r, nil
+}
+
+func getObjectMeta(ctx context.Context, v interface{}) (metav1.ObjectMeta, error) {
+	cluster, err := key.ToCluster(v)
+	if err != nil {
+		return metav1.ObjectMeta{}, microerror.Mask(err)
+	}
+
+	return metav1.ObjectMeta{
```

Comment:
Do we need to add labels like.

```
clusterID
organization
Installation 
etc? 
```

Reply:
What would you configure here?

Reply:
I'd like to have this merged so getting ping pong will not help :p

Reply:
Maybe, we need it to distinguish the created secrets.
Or we need it to implement cleanup mechanism (same we did in remotewrite secret).
But, i don't know, this was my suggestion.

Apart from that, what does `ping pong` means in this context 😄

Reply:
There is no copy of the secret so we do not need a copy mechanism as the secrets are created in the clsuter_id-prometheus namespace and they are in different namespaces so ther eis no need to differenciate them :)

Ping pong means I have to ask for more info when you could have added it on the initial review so we avoid back and forth to get to what you think is best in the end :) 

For example, you could have said something along the lines of:
Do we need to add some labels to this secret if we need to distinguish it from the other secrets (e.g. if they are in the same namespace) ?
That would have helped me get a better understanding of why you might think those labels could be needed :)

Reply:
thanks

Reply:
Done